doc/sphinx/Pacemaker_Administration/upgrading.rst

```diff
@@ -0,0 +1,485 @@
+|                                                   |          |          |        |         |          |          |
+| .. index::                                        |          |          |        |         |          |          |
+|    pair: cluster; upgrade with shutdown           |          |          |        |         |          |          |
+|    pair: upgrade; upgrade with shutdown           |          |          |        |         |          |          |
++---------------------------------------------------+----------+----------+--------+---------+----------+----------+
+| Rolling (node by node)                            | no       | yes      | always | yes     | yes      | no       |
+|                                                   |          |          | [#]_   |         |          |          |
+|                                                   |          |          |        |         |          |          |
+| .. index::                                        |          |          |        |         |          |          |
+|    pair: cluster; rolling upgrade                 |          |          |        |         |          |          |
+|    pair: upgrade; rolling upgrade                 |          |          |        |         |          |          |
++---------------------------------------------------+----------+----------+--------+---------+----------+----------+
+| Detach and reattach                               | yes      | no       | only   | no      | no       | yes      |
+|                                                   |          |          | due to |         |          |          |
+|                                                   |          |          | failure|         |          |          |
+|                                                   |          |          |        |         |          |          |
+| .. index::                                        |          |          |        |         |          |          |
+|    pair: cluster; upgrade with detach and reattach|          |          |        |         |          |          |
+|    pair: upgrade; upgrade with detach and reattach|          |          |        |         |          |          |
++---------------------------------------------------+----------+----------+--------+---------+----------+----------+
+
+Complete Cluster Shutdown
+_________________________
+
+In this scenario, one shuts down all cluster nodes and resources,
+then upgrades all the nodes before restarting the cluster.
+
+#. On each node:
+
+   #. Shutdown the cluster software (pacemaker and the messaging layer).
+   #. Upgrade the Pacemaker software. This may also include upgrading the
+      messaging layer and/or the underlying operating system.
+   #. Check the configuration with the ``crm_verify`` tool.
+
+#. On each node:
+
+   #. Start the cluster software.
+
```

Comment:
These sublists would be easier to read if they used letters instead of numbers.  That's how the old docs worked, at least.

Reply:
Didn't know that was possible -- fixed